app/MacPorts/config.py

```diff
@@ -0,0 +1,10 @@
+import os
+
+# configuration for rsync
+RSYNC = '/usr/bin/rsync'
+RSYNC_SOURCE = "rsync://rsync.macports.org/macports//trunk/dports/PortIndex_darwin_18_i386/PortIndex.json"
+JSON_FILE = os.path.join(os.path.abspath(os.path.dirname(__name__)), 'portindex.json')
```

Comment:
* Call stuff after contents, not just by type, unless it's a super local variable. Something like `PORTINDEX_JSON` maybe.
* If you do it this way, use a reasonable name for `os.path.abspath(os.path.dirname(__name__))` and do the `os.path.join` between that variable and `portindex.json`. I would probably create a separate folder outside of the sources themselves and store git clones and stuff like that all at that same place. It can be `data` next to `app`, it doesn't need to be stored in another universe, but a path like the one above definitely sounds strange.

Reply:
Thank you, idea of a `data` folder would make things neat.

Reply:
I have made suggested changes.